mita/util.py

```diff
@@ -350,3 +352,35 @@ def delete_provider_node(provider, name):
         logger.info("Node does not exist in cloud provider, cannot delete")
     except Exception:
         logger.exception("encountered errors while trying to delete node from cloud provider")
+
+
+def match_node_from_job_config(job_url):
+    config_url = os.path.join(job_url, "config.xml")
+    logger.info("Getting job config from: %s", config_url)
+    response = requests.get(config_url, auth=(conf.jenkins['user'], conf.jenkins['token']))
+    # if retrieving the config.xml fails, raise an error
+    response.raise_for_status()
```

Comment:
This would make the whole task avoid processing further.

wouldn't it be better to log the error and then `return None` if that is the expectation from the task?